### ISSUE TITLE
Categorical Predictors

### ISSUE THREAD

**Issue by oldoc63** [oldoc63/learningDS#473]

**Comment by oldoc63** [comment]
We see that people who play basketball tend to be taller than people who do not. Just like before, we can draw a line to fit these points. The best line for this plot is the one that goes through the mean height for each group. To recreate the scatter plot with the best fit line, we could use the following code:

``````
#Calculate group means
print(data.groupby('play_bball').mean().height)
``````
``````
play_bball
--
0 | 169.016
1 | 183.644
``````

**Comment by oldoc63** [comment]
1. Using the dataset students, plot a scatter plot of score (y-axis) against breakfast (x-axis) to see scores for students who did and did not eat breakfast.

**Comment by oldoc63** [comment]
2. Calculate the mean test score for students who ate breakfast and the mean score for students who did not eat breakfast. Use these numbers to plot the best-fit line on top of the scatter plot.

**Comment by oldoc63** [comment]
#### Fit and Interpretation
Now that we've seen what a regression model with a binary predictor looks like visually, we can actually fit the model using `statsmodels.api.OLS.from_formula()`, the same way we did for a quantitative predictor:

``````
model = sm.OLS.from_formula('height ~ play_bball', data)
results = model.fit()
print(results.params)
``````
``````
Intercept     169.016
play_bball     14.628
dtype: float64
``````
Note that this will work if the play_bball variable is coded with 0 and 1, but it will also work if it is coded with True and False, or even if it is coded with strings like 'yes' and 'no' (in this case, the coefficient label will look something like `play_bball[T.yes]` in the params output, indicating that 'yes' corresponds to a 1).

To interpret this output, we first need to remember that the intercept is the expected value of the outcome variable when the predictor is equal to zero. In this case, the intercept is therefore the mean height of non-basketball players.

The slope is the expected difference in the outcome variable for a one unit difference in the predictor variable. In this case, a one unit difference in play_bball is the difference between not being a basketball player and being a basketball player. Therefore, the slope is the difference in mean heights for basketball players and non-basketball players.

**Comment by oldoc63** [comment]
1. Create and fit a regression model of score predicted by breakfast using sm.OLS.from_formula() and print out the coefficients.

**Comment by oldoc63** [comment]
2. Calculate the mean test score for students who ate breakfast and the mean score for students who did not eat breakfast. Calculate and print the difference in mean scores. How this number relates to the regression output?
